examples/dishwasher-app/silabs/src/ElectricalEnergyMeasurementInstance.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/EventLogging.h>
+#include <app/reporting/reporting.h>
+#include <app/server/Server.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/ClusterEnums.h>
+#include <system/SystemClock.h>
+
+#include "AppConfig.h"
+#include "AppTask.h"
+#include "ElectricalEnergyMeasurementInstance.h"
+#include "EnergyTimeUtils.h"
+
+#define mWms_TO_mWh(power) ((power) / 3600'000)
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::DeviceEnergyManagement;
+using namespace chip::app::Clusters::ElectricalEnergyMeasurement;
+using namespace chip::app::Clusters::ElectricalEnergyMeasurement::Attributes;
+using namespace chip::app::Clusters::ElectricalEnergyMeasurement::Structs;
+using namespace chip::app::DataModel;
+
+// Example of setting CumulativeEnergyReset structure - for now set these to 0
+// but the manufacturer may want to store these in non volatile storage for timestamp (based on epoch_s)
+// Create an accuracy entry which is between +/-0.5 and +/- 5% across the range of all possible energy readings
+static const MeasurementAccuracyRangeStruct::Type kEnergyAccuracyRanges[] = {
+    { .rangeMin   = 0,
+      .rangeMax   = 1'000'000'000'000'000, // 1 million Mwh
+      .percentMax = MakeOptional(static_cast<chip::Percent100ths>(500)),
+      .percentMin = MakeOptional(static_cast<chip::Percent100ths>(50)) }
+};
+
+static const MeasurementAccuracyStruct::Type kAccuracy = {
+    .measurementType  = MeasurementTypeEnum::kElectricalEnergy,
+    .measured         = true,
```

Comment:
suggest making this False or make a comment to say real appliances should connect this data to a real metering device?